diff --git a/tsk/fs/ffs_dent.c b/tsk/fs/ffs_dent.c
index a9e9aec3..72ef538e 100644
--- a/tsk/fs/ffs_dent.c
+++ b/tsk/fs/ffs_dent.c
@@ -33,13 +33,15 @@ static uint8_t
 ffs_dent_copy(FFS_INFO * ffs, char *ffs_dent, TSK_FS_NAME * fs_name)
 {
     TSK_FS_INFO *a_fs = &(ffs->fs_info);
+    ffs_dentry1 *dir1 = NULL;
+    ffs_dentry2 *dir2 = NULL;
 
     /* this one has the type field */
     if ((a_fs->ftype == TSK_FS_TYPE_FFS1)
         || (a_fs->ftype == TSK_FS_TYPE_FFS2)) {
-        ffs_dentry1 *dir = (ffs_dentry1 *) ffs_dent;
+        dir1 = (ffs_dentry1 *) ffs_dent;
 
-        fs_name->meta_addr = tsk_getu32(a_fs->endian, dir->d_ino);
+        fs_name->meta_addr = tsk_getu32(a_fs->endian, dir1->d_ino);
 
         if (fs_name->name_size != FFS_MAXNAMLEN) {
             if (tsk_fs_name_realloc(fs_name, FFS_MAXNAMLEN))
@@ -47,9 +49,9 @@ ffs_dent_copy(FFS_INFO * ffs, char *ffs_dent, TSK_FS_NAME * fs_name)
         }
 
         /* ffs null terminates so we can strncpy */
-        strncpy(fs_name->name, dir->d_name, fs_name->name_size);
+        strncpy(fs_name->name, dir1->d_name, fs_name->name_size);
 
-        switch (dir->d_type) {
+        switch (dir1->d_type) {
         case FFS_DT_REG:
             fs_name->type = TSK_FS_NAME_TYPE_REG;
             break;
@@ -81,9 +83,9 @@ ffs_dent_copy(FFS_INFO * ffs, char *ffs_dent, TSK_FS_NAME * fs_name)
         }
     }
     else if (a_fs->ftype == TSK_FS_TYPE_FFS1B) {
-        ffs_dentry2 *dir = (ffs_dentry2 *) ffs_dent;
+        dir2 = (ffs_dentry2 *) ffs_dent;
 
-        fs_name->meta_addr = tsk_getu32(a_fs->endian, dir->d_ino);
+        fs_name->meta_addr = tsk_getu32(a_fs->endian, dir2->d_ino);
 
         if (fs_name->name_size != FFS_MAXNAMLEN) {
             if (tsk_fs_name_realloc(fs_name, FFS_MAXNAMLEN))
@@ -91,7 +93,7 @@ ffs_dent_copy(FFS_INFO * ffs, char *ffs_dent, TSK_FS_NAME * fs_name)
         }
 
         /* ffs null terminates so we can strncpy */
-        strncpy(fs_name->name, dir->d_name, fs_name->name_size);
+        strncpy(fs_name->name, dir2->d_name, fs_name->name_size);
 
         fs_name->type = TSK_FS_NAME_TYPE_UNDEF;
     }
@@ -117,6 +119,7 @@ ffs_dent_parse_block(FFS_INFO * ffs, TSK_FS_DIR * fs_dir, uint8_t a_is_del,
     unsigned int idx;
     unsigned int inode = 0, dellen = 0, reclen = 0;
     unsigned int minreclen = 4;
+    unsigned int namelen = 0;
     TSK_FS_INFO *fs = &(ffs->fs_info);
 
     char *dirPtr;
@@ -129,7 +132,7 @@ ffs_dent_parse_block(FFS_INFO * ffs, TSK_FS_DIR * fs_dir, uint8_t a_is_del,
      ** recorded length so we can view the deleted entries
      */
     for (idx = 0; idx <= len - FFS_DIRSIZ_lcl(1); idx += minreclen) {
-        unsigned int namelen = 0;
+        namelen = 0;
 
         dirPtr = (char *) &buf[idx];
 
@@ -233,6 +236,10 @@ ffs_dir_open_meta(TSK_FS_INFO * a_fs, TSK_FS_DIR ** a_fs_dir,
     char *dirbuf;
     int nchnk, cidx;
     TSK_FS_DIR *fs_dir;
+    TSK_OFF_T offset;
+    int len;
+    ssize_t cnt;
+    TSK_FS_NAME *fs_name;
 
     /* If we get corruption in one of the blocks, then continue processing.
      * retval_final will change when corruption is detected.  Errors are
@@ -240,7 +247,6 @@ ffs_dir_open_meta(TSK_FS_INFO * a_fs, TSK_FS_DIR ** a_fs_dir,
     TSK_RETVAL_ENUM retval_tmp;
     TSK_RETVAL_ENUM retval_final = TSK_OK;
 
-
     if (a_addr < a_fs->first_inum || a_addr > a_fs->last_inum) {
         tsk_error_reset();
         tsk_error_set_errno(TSK_ERR_FS_WALK_RNG);
@@ -299,11 +305,11 @@ ffs_dir_open_meta(TSK_FS_INFO * a_fs, TSK_FS_DIR ** a_fs_dir,
      */
     nchnk = (int) (size) / (FFS_DIRBLKSIZ) + 1;
 
-    TSK_OFF_T offset = 0;
+    offset = 0;
     for (cidx = 0; cidx < nchnk && (int64_t) size > 0; cidx++) {
-        int len = (FFS_DIRBLKSIZ < size) ? FFS_DIRBLKSIZ : (int) size;
+        len = (FFS_DIRBLKSIZ < size) ? FFS_DIRBLKSIZ : (int) size;
 
-        ssize_t cnt = tsk_fs_file_read(fs_dir->fs_file, offset, dirbuf, len, (TSK_FS_FILE_READ_FLAG_ENUM)0);
+        cnt = tsk_fs_file_read(fs_dir->fs_file, offset, dirbuf, len, (TSK_FS_FILE_READ_FLAG_ENUM)0);
         if (cnt != len) {
             tsk_error_reset();
             tsk_error_set_errno(TSK_ERR_FS_FWALK);
@@ -334,7 +340,7 @@ ffs_dir_open_meta(TSK_FS_INFO * a_fs, TSK_FS_DIR ** a_fs_dir,
 
     // if we are listing the root directory, add the Orphan directory entry
     if (a_addr == a_fs->root_inum) {
-        TSK_FS_NAME *fs_name = tsk_fs_name_alloc(256, 0);
+        fs_name = tsk_fs_name_alloc(256, 0);
         if (fs_name == NULL)
             return TSK_ERR;
 
